activation/nipost.go

```diff
@@ -423,18 +452,19 @@ func (nb *NIPostBuilder) submitPoetChallenges(
 	}
 	if allInvalid {
 		nb.logger.Warn("all poet submits were too late. ATX challenge expires", log.ZShortStringer("smesherID", nodeID))
-		return ErrATXChallengeExpired
+		return nil, ErrATXChallengeExpired
 	}
-	return nil
-}
 
-func (nb *NIPostBuilder) getPoetService(ctx context.Context, address string) PoetService {
-	for _, service := range nb.poetProvers {
-		if address == service.Address() {
-			return service
-		}
+	existingRegistrations, err = nipost.PoetRegistrationsByNodeIdAndAddresses(nb.localDB, signer.NodeID(), addresses)
```

Comment:
Quierying the DB again could be avoided here. There is already a list of `existingRegistrations`, which could be extended with the successful registrations obtained from the individual calls to `nb.submitPoetChallenge` (if it returned a _registration object_). Wdyt?